### ISSUE TITLE
[Bug]: Attempt to uninstall and reinstall devbox failed

### ISSUE THREAD

**Issue by cschoreisz-pillar** [jetpack-io/devbox#1712]
**Current Behavior (bug)**
Upon upgrading to latest version, nix started throwing errors, and I could not do `devbox shell` anymore.
I then tried to fully uninstall devbox following [these instructions](https://gist.github.com/bxb100/da14db262c040b2f4d16cd90f6146e2e) (complemented by [How to remove Nix](https://github.com/NixOS/nix/issues/1402#issuecomment-312496360) and [this comment](https://github.com/NixOS/nix/issues/1402#issuecomment-716915515) as well) 
- uninstall devbox
- erase then remove Nix Store from disk utilities
- remove lingering nix files and folders
- empty bin
- delete all `nixbld0`, `nixbld1`, ... `nixbldn` users
- restart laptop
- reinstall devbox

Upon reinstalling and running `devbox shell` for the first time, I got this error message:
```
...
install: mkdir /etc/nix

~~> Installing Nix
      Alright! We have our first nix at /nix/store/6zi5fqzn9n17wrk8r41rhdw4j7jqqsi3-nix-2.18.1
      Just finished getting the nix database ready.

~~> Setting up shell profiles: /etc/bashrc /etc/profile.d/nix.sh /etc/zshrc /etc/bash.bashrc /etc/zsh/zshrc

# Nix
if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
  . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
fi
# End Nix


# Nix
if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
  . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
fi
# End Nix


# Nix
if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
  . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
fi
# End Nix


~~> Setting up shell profiles for Fish with conf.d/nix.fish inside /etc/fish /usr/local/etc/fish /opt/homebrew/etc/fish /opt/local/etc/fish

~~> Setting up the default profile
installing 'nix-2.18.1'
error: the build users group 'nixbld' has no members

---- oh no! --------------------------------------------------------------------
Oh no, something went wrong. If you can take all the output and open
an issue, we'd love to fix the problem so nobody else has this issue.

:(

We'd love to help if you need it.

You can open an issue at
https://github.com/NixOS/nix/issues/new?labels=installer&template=installer.md

Or get in touch with the community: https://nixos.org/community
Error: exit status 1

Error: There was an internal error. Run with DEVBOX_DEBUG=1 for a detailed error message, and consider reporting it at https://github.com/jetpack-io/devbox/issues
```

**Expected Behavior (fix)**
I would like to successfully run `devbox shell`

**Additional context**
- OS: macOS Monterrey 12.5.1, intelCore i9
- devbox version 0.8.5
- devbox.json:
```
{
  "packages": [
    "nodejs@18",
    "nodePackages.pnpm",
    "postgresql",
    "nodePackages.typescript-language-server",
    "nodePackages.prettier",
    "colima"
  ],
  "nixpkgs": {
    "commit": "f80ac848e3d6f0c12c52758c0f25c10c97ca3b62"
  },
  "shell": {
    "scripts": {
      "pnpm-store-path": "pnpm store path",
      "install": "pnpm i",
      "test": "pnpm turbo test:unit"
    }
  }
}
```

**Comment by savil** [comment]
yikes!

> Upon upgrading to latest version

were you upgrading to the latest version of Devbox, or of nix?

**Comment by cschoreisz-pillar** [comment]
Hey @savil I was updating to the latest version of devbox

**Comment by cschoreisz-pillar** [comment]
I did a second attempt and it worked ! 
I did 2 things differently this time:
- I emptied my bin (it was specified at the bottom of the instructions)
- I removed all `nixbld` users which I thought I had already done, but turned out it hadn’t worked.

Food for thoughts 🌱 : It'd be great for the instructions' steps to be fully laid out in an easy to find documentation, instead of this [tiny guide](https://gist.github.com/bxb100/da14db262c040b2f4d16cd90f6146e2e) redirecting to 3 important links, each of them needing to be carefully followed.

**Comment by savil** [comment]
@cschoreisz-pillar sorry to hear about this poor experience, and glad you are unblocked. Would you happen to know what versions of nix and Devbox you had prior to starting the upgrade?

**Comment by cschoreisz-pillar** [comment]
All good @savil, it's always an occasion for me to learn, and I appreciated the fast answers.

I was on 0.5.11 for devbox, and I dont know about the nix version associated to it unfortunately

**Comment by savil** [comment]
@cschoreisz-pillar Thanks for sharing the version info.

We try hard to ensure Devbox version updates are backwards compatible. That said, the times we have failed to ensure that have been around Devbox's internal logic, rather than around the underlying nix system. Its very concerning to me that you had to resort to the drastic measure of re-installing nix. That is not normal at all.

I hope this won't discourage you from updating Devbox more frequently. If you do encounter similar issues next time, please save the output somewhere and file a github issue or hit us up on Discord.
